

# Quiz (19.09.23)


```{r}
library(tidyverse, quietly = T)
library(readr, quietly = T)
```

```{r}
cols = c('chrom', 'source', 'feature_type', 'start', 'end', 'score', 'strand', 'phase', 'info')
d <- read_delim('gencode.v31.basic.annotation.gtf.gz', 
               delim='\t', skip = 5, progress = F, 
               col_names = cols)
```



```{r}
head(d)
```



1. Calculate the number of genes per chromosome.

```{r}
d1 <- filter(d, feature_type == "gene")
d2 <- d1 %>% 
  group_by(chrom) %>%
  count(feature_type)
d2
```


2. Compare the number of genes between autosomal and sex chromosome. (Mean, Median)
```{r}
d_gene <- d1 %>%
  mutate(isSexChrome = ifelse(chrom %in% c('chrX', 'chrY'), 'sex_chromosome', 'autosome'))
d_gene %>%
  group_by(isSexChrome) %>%
  summarize(Mean = mean(d2$n), Median = median(d2$n))
```


3. Divide the genes into groups 'protein coding' and 'long noncoding', and compare the number of genes in each chromosomes within groups.
```{r}
d_split <- strsplit(d1$info, split="; gene_")
d1 %>%
  mutate(group = do.call(rbind.data.frame, d_split)[,2]) %>%
  filter(group == 'type "protein_coding"' | group == 'type "lncRNA"') %>%
  group_by(chrom, group) %>%
  count(feature_type)
```

